File: app/src/main/java/net/schueller/peertube/network/AuthorizationInterceptor.java
/*
 * Copyright (C) 2020 Stefan Schüller <sschueller@techdroid.com>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package net.schueller.peertube.network;

import android.util.Log;

import androidx.annotation.NonNull;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class AuthorizationInterceptor implements Interceptor {

    public AuthorizationInterceptor() {
    }

    @NonNull
    @Override
    public Response intercept(Chain chain) throws IOException {

        Session session = Session.getInstance();
        Response mainResponse;
        Request mainRequest = chain.request();

        if (session.isLoggedIn()) {

            // add authentication header to each request if we are logged in
            Request.Builder builder = mainRequest.newBuilder().header("Authorization", session.getToken()).
                    method(mainRequest.method(), mainRequest.body());
           // Log.v("Authorization", "Intercept: " + session.getToken());

            // build request
            Request req = builder.build();
            mainResponse = chain.proceed(req);

            // logout on auth error
            if (mainResponse.code() == 401 || mainResponse.code() == 403) {
                session.invalidate();
                Log.v("Authorization", "Intercept: Logout forced");
            }

        } else {
            mainResponse = chain.proceed(chain.request());
        }

        return mainResponse;
    }

}

File: app/src/main/java/net/schueller/peertube/feature_video/data/remote/auth/AuthorizationInterceptor.kt
package net.schueller.peertube.feature_video.data.remote.auth

import android.util.Log
import okhttp3.Interceptor
import okhttp3.Response
import okhttp3.Request
import javax.inject.Inject

class AuthorizationInterceptor @Inject constructor(
    private val session: Session
) :Interceptor {
    override fun intercept(chain: Interceptor.Chain): Response {

        val tag = "AInterc"

        val mainResponse: Response
        val mainRequest: Request = chain.request()
        val token = session.getToken()

        if (session.isLoggedIn() && token !== null) {

            // add authentication header to each request if we are logged in
            val builder: Request.Builder =
                mainRequest.newBuilder().header("Authorization", token)
                    .method(mainRequest.method, mainRequest.body)
            Log.v(tag, "Intercept: " + session.getToken());

            // build request
            val req: Request = builder.build()
            mainResponse = chain.proceed(req)

            // logout on auth error
            if (401 or 403 == mainResponse.code) {
                session.invalidate()
                Log.v(tag, "Intercept: Logout forced")
            }
        } else {
            Log.v(tag, "Mot logged in or no token")
            mainResponse = chain.proceed(chain.request())
        }

        return mainResponse

    }

}